app/src/main/java/org/wikipedia/descriptions/DescriptionEditFragment.kt

```diff
@@ -0,0 +1,442 @@
+                    }
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({ result ->
+                        if (result.hasEditResult() && result.edit() != null) {
+                            when {
+                                result.edit()!!.editSucceeded() -> {
+                                    Handler().postDelayed(successRunnable, TimeUnit.SECONDS.toMillis(4))
+                                    funnel.logSaved(result.edit()!!.newRevId())
+                                }
+                                result.edit()!!.hasEditErrorCode() -> {
+
+                                    // TODO: handle AbuseFilter messages
+                                    // new EditAbuseFilterResult(result.edit().code(), result.edit().info(), result.edit().warning());
+                                    editFailed(IOException(StringUtils.defaultString(result.edit()!!.warning())), false)
+                                }
+                                result.edit()!!.hasCaptchaResponse() -> {
+
+                                    // TODO: handle captcha.
+                                    // new CaptchaResult(result.edit().captchaId());
+                                    funnel.logCaptchaShown()
+                                }
+                                result.edit()!!.hasSpamBlacklistResponse() -> {
+                                    editFailed(IOException(getString(R.string.editing_error_spamblacklist)), false)
+                                }
+                                else -> {
+                                    editFailed(IOException("Received unrecognized edit response"), true)
+                                }
+                            }
+                        } else {
+                            editFailed(IOException("An unknown error occurred."), true)
+                        }
+                    }) { caught -> editFailed(caught, true) })
+        }
+
+        private fun postDescriptionToWikidata(editToken: String) {
+            disposables.add(ServiceFactory.get(WikiSite.forLanguageCode(pageTitle!!.wikiSite.languageCode())).siteInfo
+                    .flatMap { response ->
+                        val languageCode = if (response.query()!!.siteInfo() != null && response.query()!!.siteInfo()!!.lang != null &&
+                                response.query()!!.siteInfo()!!.lang != AppLanguageLookUpTable.CHINESE_LANGUAGE_CODE) response.query()!!.siteInfo()!!.lang
+                        else pageTitle!!.wikiSite.languageCode()
+                        getPostObservable(editToken, languageCode!!)
+                    }
+                    .subscribeOn(Schedulers.io())
+                    .observeOn(AndroidSchedulers.mainThread())
+                    .subscribe({ response ->
+                        if (response.successVal > 0) {
+                            Handler().postDelayed(successRunnable, TimeUnit.SECONDS.toMillis(4))
```

Comment:
since `Handler()` is deprecated, let's switch to using `requireView.postDelayed()`.